src/measure_theory/measure/finite_measure_weak_convergence.lean

```diff
@@ -0,0 +1,246 @@
+end
+
+lemma to_measure_eq_val (ν : finite_measure α) : (ν : measure α) = ν.val := rfl
+
+lemma coe_injective :
+  function.injective (coe : finite_measure α → measure α) :=
+subtype.coe_injective
+
+@[simp]
+lemma coe_zero : (coe : finite_measure α → measure α) 0 = 0 := rfl
+
+@[simp] lemma coe_add (μ ν : finite_measure α) : ↑(μ + ν) = (↑μ + ↑ν : measure α) := rfl
+
+@[simp] lemma coe_smul (c : ℝ≥0) (μ : finite_measure α) : ↑(c • μ) = (c • ↑μ : measure α) := rfl
+
+/-- The (total) mass of a finite measure `μ` is `μ univ`, i.e., the cast to `nnreal` of
+`(μ : measure α) univ`. -/
+def mass (μ : finite_measure α) : ℝ≥0 := μ univ
+
+@[simp] lemma mass_ennreal {μ : finite_measure α} :
+  (μ.mass : ℝ≥0∞) = (μ : measure α) univ :=
+begin
+  apply ennreal.coe_to_nnreal,
+  exact ennreal.lt_top_iff_ne_top.mp (μ.prop).measure_univ_lt_top,
+end
+
+-- TODO: Another alternative would use a bijection with `vector_measure α ℝ≥0`.
+instance : add_comm_monoid (finite_measure α) :=
+(finite_measure.coe_injective).add_comm_monoid
+  (coe : finite_measure α → measure α) finite_measure.coe_zero finite_measure.coe_add
+
+/-- Coercion is an `add_monoid_hom`. -/
+@[simps]
+def coe_add_monoid_hom : finite_measure α →+ measure α :=
+{ to_fun := coe, map_zero' := coe_zero, map_add' := coe_add }
+
+-- TODO: Another alternative would use a bijection with `vector_measure α ℝ≥0`.
+instance {α : Type*} [measurable_space α] : module ℝ≥0 (finite_measure α) :=
+function.injective.module _ coe_add_monoid_hom finite_measure.coe_injective coe_smul
+
+end finite_measure -- end namespace
+
+end finite_measure -- end section
+
+section probability_measure
+/-!
+### Probability measures
+In this section, we define the `Type` of probability measures on a given measurable space. This
+type will be equipped with the topology of weak convergence of measures when the measurable space
+is a topological space equipped with its Borel sigma-algebra.
+
+There is a coercion (embedding) to finite measures on the same space.
+-/
+
+variables {α : Type} [measurable_space α]
+
+/-- Probability measures are defined as the subtype of measures that have the property of being
+probability measures (i.e., their total mass is one). -/
+def probability_measure (α : Type) [measurable_space α] : Type :=
+{μ : measure α // is_probability_measure μ}
+
+namespace probability_measure
+
+instance [inhabited α] : inhabited (probability_measure α) :=
+{ default := ⟨measure_theory.measure.dirac (default α),
+              measure_theory.measure.dirac.is_probability_measure⟩ }
+
+/-- A probability measure can be interpreted as a measure. -/
+instance : has_coe (probability_measure α) (measure_theory.measure α) := coe_subtype
+
+instance : has_coe_to_fun (probability_measure α) :=
+⟨λ _, set α → ℝ≥0, λ μ s, (μ s).to_nnreal⟩
+
+instance to_measure.is_probability_measure (μ : probability_measure α) :
+  is_probability_measure (μ : measure α) := μ.prop
```

Comment:
I think it would be fine to let this instance be auto-named:
```suggestion
instance (μ : probability_measure α) : is_probability_measure (μ : measure α) := μ.prop
```
(or did you try this and find yourself unhappy with the auto-name?)

Reply:
I think I again simply didn't realize that the auto-naming is so good. (...despite being told so in the previous PR...)